File: src/main/java/com/snooper/service/PlateauNavigation.java
package com.snooper.service;

import com.snooper.constant.Action;
import com.snooper.dto.Plateau;
import com.snooper.dto.RoboticRover;
import com.snooper.exception.OutOfBoundaryException;

import java.util.List;

public interface PlateauNavigation {

    void handle(Plateau plateau, RoboticRover roboticRover, List<Action> actions) throws OutOfBoundaryException;
}

File: src/main/java/com/snooper/service/PlateauNavigation.kt
package com.snooper.service

import com.snooper.constant.Action
import com.snooper.dto.Plateau
import com.snooper.dto.RoboticRover
import com.snooper.exception.OutOfBoundaryException

interface PlateauNavigation {
    @Throws(OutOfBoundaryException::class)
    fun handle(plateau: Plateau, roboticRover: RoboticRover, actions: List<Action>)
}